src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/KeyStatisticService.cs

```diff
@@ -0,0 +1,192 @@
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess(async release =>
+                await _persistenceHelper.CheckEntityExists<KeyStatistic>(keyStatisticId, query =>
+                    query
+                        .Include(keyStat => (keyStat as KeyStatisticDataBlock)!.DataBlock) // TODO Include can be removed in EES-3988
+                        .Where(keyStat => keyStat.ReleaseId == release.Id)))
+            // TODO: Can remove the below OnSuccess in EES-3988
+            // Key stats were previously associated with a KeyStatistic content section,
+            // which meant their data blocks had a ContentSectionId set. The below code
+            // ensures ContentSectionId is unset when a key stat is removed. If it isn't
+            // unset, that data block won't become available for selection for content again.
+            .OnSuccess(async keyStat =>
+            {
+                if (keyStat is KeyStatisticDataBlock keyStatDataBlock)
+                {
+                    var dataBlock = keyStatDataBlock.DataBlock;
+                    dataBlock.ContentSection = null;
+                    dataBlock.ContentSectionId = null;
+                    _context.ContentBlocks.Update(dataBlock);
+                    await _context.SaveChangesAsync();
+                }
+                return keyStat;
+            })
+            .OnSuccessVoid(async keyStat =>
+            {
+                _context.KeyStatistics.Remove(keyStat);
+                await _context.SaveChangesAsync();
+            });
+    }
+
+    public async Task DeleteAssociatedKeyStatisticDataBlock(Guid dataBlockId)
+    {
+        var keyStat = await _context
+            .KeyStatisticsDataBlock
+            .SingleOrDefaultAsync(ks => ks.DataBlockId == dataBlockId);
+
+        if (keyStat != null)
+        {
+            _context.KeyStatisticsDataBlock.RemoveRange(keyStat);
+            await _context.SaveChangesAsync();
+        }
+    }
+
+    public async Task<Either<ActionResult, List<KeyStatisticViewModel>>> Reorder(
+        Guid releaseId,
+        List<Guid> newOrder)
+    {
+        return await _persistenceHelper.CheckEntityExists<Release>(releaseId)
+            .OnSuccess(_userService.CheckCanUpdateRelease)
+            .OnSuccess<ActionResult, Release, List<KeyStatisticViewModel>>(async release =>
+            {
+                var keyStatistics = await _context.KeyStatistics
+                    .Where(ks => ks.ReleaseId == release.Id)
+                    .ToListAsync();
+
+                if (!ComparerUtils.SequencesAreEqualIgnoringOrder(
+                        newOrder,
+                        keyStatistics.Select(ks => ks.Id)))
+                {
+                    return ValidationUtils.ValidationActionResult(
+                        ProvidedKeyStatIdsDifferFromReleaseKeyStatIds);
+                }
+
+                var updatingUserId = _userService.GetUserId();
```

Comment:
Not sure there's much point assigning this as a variable.

Reply:
Discussed on Slack - going to leave it. It prevents `GetUserId` being called multiple times in the `ForEach` loop.